packages/grid/x-data-grid/src/models/api/gridCallbackDetails.ts

```diff
@@ -1,10 +1,16 @@
+import { GridControlledStateReasonLookup } from '../events/gridEventLookup';
+
 /**
  * Additional details passed to the callbacks
  */
-export interface GridCallbackDetails {
+export interface GridCallbackDetails<K extends keyof GridControlledStateReasonLookup = any> {
```

Comment:
Once `apiRef.current.unstable_updateControlState` stops being only a wrapper for `setState` we can replace the `any` with the correct types. The `any` is necessary because we don't know which key of the state was changed to match with `GridControlledStateReasonLookup`.